tls/s2n_prf.h

```diff
@@ -26,21 +26,23 @@
 /* Enough to support TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384, 2*SHA384_DIGEST_LEN + 2*AES256_KEY_SIZE */
 #define S2N_MAX_KEY_BLOCK_LEN 160
 
-struct p_hash_state {
+/* TODO: make this a union. If s2n is in FIPS mode it's always going to use evp_hmac, and outside of FIPS it should
+ * always use s2n_hmac */
+struct hmac_state {
     struct s2n_hmac_state s2n_hmac;
     struct s2n_evp_hmac_state evp_hmac;
 };
```

Comment:
This name is going to be potentially confusing, given: https://github.com/aws/s2n-tls/blob/main/crypto/s2n_hmac.h#L34-L52

Reply:
Since this is in a module called "s2n_prf" and is only used for the PRF, maybe work PRF into the name? prf_state?

Reply:
I decided to keep the current p_hash naming convention because this is the specific HMAC implementation to be used in the P_hash function.